cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -220,18 +220,60 @@ pub enum Inst {
         srcloc: Option<SourceLoc>,
     },
 
-    /// XMM (scalar) unary op (from xmm to integer reg): movd, movq
+    /// XMM (scalar) unary op (from xmm to integer reg): movd, movq, cvtts{s,d}2si
     XmmToGpr {
         op: SseOpcode,
         src: Reg,
         dst: Writable<Reg>,
+        dst_size: OperandSize,
     },
 
-    /// XMM (scalar) unary op (from integer to float reg): movd, movq
+    /// XMM (scalar) unary op (from integer to float reg): movd, movq, cvtsi2s{s,d}
     GprToXmm {
         op: SseOpcode,
         src: RegMem,
         dst: Writable<Reg>,
+        src_size: OperandSize,
+    },
+
+    /// Converts an unsigned int64 to a float64.
+    CvtUint64ToFloatSeq {
+        /// Is the target a 64-bits or 32-bits register?
+        to_f64: bool,
+        src: Reg,
+        dst: Writable<Reg>,
+        tmp_gpr1: Writable<Reg>,
+        tmp_gpr2: Writable<Reg>,
+    },
+
+    /// Converts a scalar xmm to a signed int32/int64.
+    CvtFloatToSintSeq {
+        dst_size: OperandSize,
+        src_size: OperandSize,
+        /// A copy of the source register, fed by lowering. It is marked as modified during
```

Comment:
We (collectively) should discuss register conventions, regarding `get_regs` and how it interacts with temp regs in these sequences, since I think we are in danger of falling into a nasty correctness hole otherwise.  Not just here; the in-progress aarch64 atomics work has the same problem.  Once we have a clear common understanding, we can go back and audit this stuff.  I'm just noting it here; doesn't need to block landing right now.